packages/core/src/auth/secondaryAuth.ts

```diff
@@ -289,7 +283,34 @@ export class SecondaryAuth<T extends Connection = Connection> {
             getLogger().warn(`auth (${this.toolId}): saved connection "${this.key}" is not valid`)
             await globalState.update(this.key, undefined)
         } else {
-            await this.auth.refreshConnectionState(conn)
+            const getAuthStatus = (state: ReturnType<typeof this.auth.getConnectionState>): AuthStatus => {
+                return state === 'invalid' ? 'expired' : 'connected'
+            }
+
+            let connectionState = this.auth.getConnectionState(conn)
+            telemetry.auth_modifyConnection.record({
```

Comment:
nit: In the case we aren't already running within an `auth_modifyConnection` run, this will do nothing.
I'm wondering if it will be useful to specify where we expect the run to come from, so in this case from `restoreConnection()`